### PR TITLE
Change all examples to TS

### PR DESCRIPTION
**Pull request by tell-y**
Hi, I've continued https://github.com/pmndrs/jotai/pull/515 for all examples.

Current examples and changes:
- hacker news
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/hacker_news
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/hacker_news

- hello
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/hello
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/hello

- mega form
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/mega-form
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/mega-form

- text length
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/text_length
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/text_length

- todos
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/todos
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/todos

- todos with atomFamily
  - https://githubbox.com/pmndrs/jotai/tree/master/examples/todos_with_atomFamily
  - https://githubbox.com/tell-y/jotai/tree/feat/examples-ts/examples/todos_with_atomFamily

Thank you for your review.

### CODE REVIEW

**Comment by dai-shi** [comment]
@Aslemammad can you review?

File: examples/hacker_news/package.json [hunk 1]
```diff
@@ -3,35 +3,30 @@
...
+  "main": "src/index.tsx",
   "dependencies": {
-    "@react-spring/web": "9.0.0-rc.3",
-    "html-react-parser": "0.14.2",
-    "jotai": "^0.15.2",
-    "react": "17.0.1",
-    "react-dom": "17.0.1",
+    "@react-spring/web": "^9.2.3",
+    "html-react-parser": "^1.2.6",
+    "jotai": "^0.16.11",
```
Comment (Aslemammad) [inline examples/hacker_news/package.json:10]: Thank you for your effort. Could you update this to jotai v1?

Comment (Aslemammad) [reply examples/hacker_news/package.json:10]: And also other examples.

Comment (tell-y) [reply examples/hacker_news/package.json]: I have done all.
@Aslemammad

Comment (Aslemammad) [reply examples/hacker_news/package.json]: Thanks.

**Comment by dai-shi** [comment]
Thanks for your work!

**Pull request merged**